### DISCUSSION TITLE
GIT - return directly to the tag

### DISCUSSION THREAD

**Discussion by Mefisteoo** [community/community#22458] [New to GitHub]
<p>Hi,<br>
I would like to go back to the specific tag that I put on the server. for example, I have four tags on the server: v1, v2, v3 and v4. I would like to go back to the v1 tag to work on it and put its changes as a v5 tag. For clarification. I made a bunch of mistakes in the v2 - v4 tags and I would like to go back to the v1 version and correct it. Is this possible?</p>
<p>Have a good day!</p>

**Comment by airtower-luna** [comment]
<p>You can check out an existing tag (e.g. your <code>v1</code>) just like a branch. I recommend creating a branch from there, e.g.:</p>
<pre><code class="lang-auto">git checkout -b for-v5 v1
</code></pre>
<p>And then you can rename your branches as you like. Alternatively you could use <a href="https://git-scm.com/docs/git-reset"><code>git reset</code></a> to reset an existing branch to another commit, but it’s much easier to break things that way (“break” as in delete the wrong thing).</p>

**Comment by Mefisteoo** [comment]
<p>I would like to stay in the ‘master’ branch.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Then you can <a href="https://git-scm.com/docs/git-reset#Documentation/git-reset.txt-emgitresetemltmodegtltcommitgt">use <code>git reset</code></a> while on that branch. Not that that is a potentially destructive operation, <code>--hard</code> overwrites the files in your worktree. :warning:</p>
<pre><code class="lang-auto">git reset --hard v1
</code></pre>
<p>I maintain it’d be safer to create a new branch. Once you’re sure that branch looks like you want it to you can delete the old <code>master</code> branch and rename the new branch.</p>
